packages/extension-chrome/src/pages/Popup/containers/Network/EditNetwork.tsx

```
@@ -0,0 +1,153 @@
+
+const HTTP_URL_PATTERN = /^https?:\/\/[a-zA-Z0-9-._/]+(:\d+)?$/;
+
+type EditNetworkFormState = {
+  name: string;
+  rpcUrl: string;
+};
+
+type EditNetworkProps = {
+  mode: 'add' | 'modify';
+};
+
+export const EditNetwork: FC<EditNetworkProps> = ({ mode }) => {
+  const navigate = useNavigate();
+  const configService = useService('configService');
+  const { id } = useParams() as { id: string };
+
+  const { handleSubmit, register, formState, setValue, trigger } = useForm<EditNetworkFormState>();
+
+  const onSubmit = handleSubmit(async ({ name, rpcUrl }, e) => {
+    e?.preventDefault();
+    if (mode === 'modify') {
+      await modifyNetworkMutation.mutateAsync({ name, rpcUrl, id });
+    }
+    if (mode === 'add') {
+      await addNetworkMutation.mutateAsync({ name, rpcUrl });
+    }
+    navigate(-1);
+  });
+
+  const modifyNetworkMutation = useMutation({
+    mutationFn: async ({ name, rpcUrl, id }: EditNetworkFormState & { id: string }) => {
+      const networks = await configService.getNetworks();
+      const newNetworks = produce(networks, (draft) => {
+        for (const network of draft) {
+          if (network.id === id) {
+            network.displayName = name;
+            network.rpcUrl = rpcUrl;
+            network.networkName = name;
+            return;
+          }
+        }
+      });
+
+      return configService.setConfig({
+        config: {
+          networks: newNetworks,
+        },
+      });
+    },
+  });
+
+  useEffect(() => {
+    if (mode === 'modify') {
+      if (id) {
+        (async () => {
+          const networks = await configService.getNetworks();
+          const network = networks.find((network) => network.id === id);
+          if (network) {
+            setValue('name', network.networkName);
+            setValue('rpcUrl', network.rpcUrl);
+            // manually trigger validation
+            await trigger();
+          }
+        })().catch(() => {});
```

we can tip for error here instead of just catching without doing anything. Or if we do want to leave an empty catch block, a good practice is to leave a comment explaining why an empty catch block is needed